include/s3select_csv_parser.h

```diff
@@ -1,413 +1,255 @@
-    a_row < In_quote_st 	, event_not_column_sep  	, In_quote_st, &csv_rules::in_quote    >,   //stay in quote
-    a_row < In_quote_st 	, event_eol  	, End_of_line_st, &csv_rules::in_quote    >,   //end of quote/line
-    a_row < In_quote_st 	, event_end_of_stream  	, End_of_line_st, &csv_rules::in_quote    >,   //end of quote/line
-
-
-  //TODO add transitions for escape just before eol , eos.
-    a_row < Start_new_token_st 	, event_escape  	, In_esc_start_token_st, &csv_rules::in_escape_start_new_token    >,
-    a_row < In_esc_start_token_st, event_column_sep, In_new_token_st, &csv_rules::in_escape    >,      //escape column-sep
-    a_row < In_esc_start_token_st, event_not_column_sep, In_new_token_st, &csv_rules::in_escape    >,
-    a_row < In_esc_start_token_st, event_escape, In_new_token_st, &csv_rules::in_escape    >,
-    a_row < In_esc_start_token_st, event_quote, In_new_token_st, &csv_rules::in_escape    >,
-
-    a_row < In_new_token_st, event_escape, In_esc_in_token_st, &csv_rules::in_escape    >,
-    a_row < In_esc_in_token_st, event_column_sep, In_new_token_st, &csv_rules::in_escape    >,
-    a_row < In_esc_in_token_st, event_not_column_sep, In_new_token_st, &csv_rules::in_escape    >,
-    a_row < In_esc_in_token_st, event_escape, In_new_token_st, &csv_rules::in_escape    >,
-    a_row < In_esc_in_token_st, event_quote, In_new_token_st, &csv_rules::in_escape    >,
-
-    a_row < In_quote_st, event_escape, In_esc_quote_st, &csv_rules::in_escape    >,
-    a_row < In_esc_quote_st, event_column_sep, In_quote_st, &csv_rules::in_escape    >,
-    a_row < In_esc_quote_st, event_not_column_sep, In_quote_st, &csv_rules::in_escape    >,
-    a_row < In_esc_quote_st, event_escape, In_quote_st, &csv_rules::in_escape    >,
-    a_row < In_esc_quote_st, event_quote, In_quote_st, &csv_rules::in_escape    >
-
-  //  +---------+-------------+---------+---------------------+----------------------+
-    > {};
-
-  // Replaces the default no-transition response.
-  template <class FSM, class Event>
-  void no_transition(Event const& e, FSM&, int state)
-  {
-    std::cout << "no transition from state " << state
-              << " on event " << typeid(e).name() << std::endl;
-  }
-}; //// end-of-state-machine
-
-
-
-// Pick a back-end
-typedef msm::back::state_machine<csvStateMch_> csvStateMch;
-
-//
-// Testing utilities.
-//
-
-static char const* const state_names[] = {"Start_new_token_st", "In_new_token_st", "In_quote_st", "In_esc_in_token_st",
-                                          "In_esc_quote_st", "In_esc_start_token_st", "End_of_line_st", "Empty_state"
-                                         };
-void pstate(csvStateMch const& p)//debug
-{
-  std::cout << " -> " << state_names[p.current_state()[0]] << std::endl;
+#include "csvparser/csv.h"
+
+namespace io{
+
+namespace detail{
+    template<class trim_policy, class quote_policy>
+                void parse_line(
```

Comment:
the s3select-engine needs a `parse_line` without the number-of-column in its interface. (`col_order`)
the engine "knows" how to handle not-specific-number of columns (skip it, or place null)

by removing `col_order`, the function will not throw an exception (too-few, too-many)
all it needs is to chop the line and return the tokens.